command/client.go

```diff
@@ -60,3 +56,16 @@ func newConsulClient(config config.ConsulConfig) (*consul.Client, error) {
 	}
 	return client, nil
 }
+
+// setEnvTemp sets an environment variable and returns a function that restores
+// the previous state.
+func setEnvTemp(k, v string) (reset func()) {
```

Comment:
This is intentionally a change in behavior. Before, the cleanup would unconditionally unset an environment variable, even if it had been set before calling `newConsulClient`. This change preserves the existing state, where the env var is unset if it was unset, zero if it was zero, and set to the original value otherwise.